src/builtins/BuiltinSharedArrayBuffer.cpp

```diff
@@ -40,27 +40,58 @@ static Value builtinSharedArrayBufferConstructor(ExecutionState& state, Value th
         ErrorObject::throwBuiltinError(state, ErrorObject::RangeError, state.context()->staticStrings().SharedArrayBuffer.string(), false, String::emptyString, ErrorObject::Messages::GlobalObject_FirstArgumentInvalidLength);
     }
 
-    return SharedArrayBufferObject::allocateSharedArrayBuffer(state, newTarget.value(), byteLength);
+    Optional<uint64_t> maxByteLength;
+    // Let requestedMaxByteLength be ? GetArrayBufferMaxByteLengthOption(options).
+    if (UNLIKELY((argc > 1) && argv[1].isObject())) {
+        Object* options = argv[1].asObject();
+        Value maxLengthValue = options->get(state, ObjectPropertyName(state.context()->staticStrings().maxByteLength)).value(state, options);
+        if (!maxLengthValue.isUndefined()) {
+            maxByteLength = maxLengthValue.toIndex(state);
+            if (UNLIKELY((maxByteLength.value() == Value::InvalidIndexValue) || (byteLength > maxByteLength.value()))) {
+                ErrorObject::throwBuiltinError(state, ErrorObject::RangeError, state.context()->staticStrings().SharedArrayBuffer.string(), false, String::emptyString, ErrorObject::Messages::GlobalObject_FirstArgumentInvalidLength);
+            }
+        }
+    }
+
+    return SharedArrayBufferObject::allocateSharedArrayBuffer(state, newTarget.value(), byteLength, maxByteLength);
 }
 
+#define RESOLVE_THIS_BINDING_TO_SHAREDARRAYBUFFER(NAME, OBJ, BUILT_IN_METHOD)                                                                                                                                                                            \
```

Comment:
If you're done using it, why don't you add `#undef` at that point?

Reply:
This macro is defined and used for this cpp file only, so it cannot affect other files.
Furthermore, this macro is used several points in this file and can be used more when other new features are added later.
So, I think that its better to maintain the current status without `#undef`.